apps/web/vibes/soul/examples/sections/cart/electric.tsx

```diff
@@ -1,24 +1,247 @@
-import { getLineItems } from '@/vibes/soul/data/line-items';
-import { Cart } from '@/vibes/soul/sections/cart';
+import { SubmissionResult } from '@conform-to/react';
+import { parseWithZod } from '@conform-to/zod';
+import Form from 'next/form';
+import { cookies } from 'next/headers';
+import { z } from 'zod';
 
-import { checkoutAction, lineItemAction } from './actions';
+import { Button } from '@/vibes/soul/primitives/button';
+import { Cart, CartLineItem } from '@/vibes/soul/sections/cart';
+import { cartLineItemActionFormDataSchema } from '@/vibes/soul/sections/cart/schema';
 
 export default async function Preview() {
-  const lineItems = await getLineItems('Electric');
-
   return (
-    <Cart
-      cart={{
-        lineItems,
-        summaryItems: [
-          { label: 'Subtotal', value: '$100' },
-          { label: 'Shipping', value: 'TBD' },
-          { label: 'Tax', value: 'TBD' },
-        ],
-        total: '127.60',
-      }}
-      checkoutAction={checkoutAction}
-      lineItemAction={lineItemAction}
-    />
+    <div className="p-4">
+      <Form action={resetCart}>
+        <Button type="submit" variant={'ghost'}>
+          Reset cart
+        </Button>
+      </Form>
+      <Cart
+        cart={await getCart()}
```

Comment:
Do we want to stream this in for the example?

Reply:
This was the one I was having the TypeScript issue with. I couldn't figure out how to get around the TypeScript issues :(

Reply:
I left a comment in Slack on how to address it by using the `Omit` utility on the `Cart` props. Let me know if you're still having trouble with it after applying that change